addons/mail/static/src/components/composer/composer_tests.js

```diff
@@ -1141,6 +1141,102 @@ QUnit.test('warning on send with shortcut when attempting to post message with s
     );
 });
 
+QUnit.test('remove an uploading attachment', async function (assert) {
+    assert.expect(4);
+
+    await this.start({
+        async mockFetch(resource, init) {
+            const res = this._super(...arguments);
+            if (resource === '/web/binary/upload_attachment') {
+                await new Promise(() => {});
+            }
+            return res;
+        }
+    });
+    const composer = this.env.models['mail.composer'].create();
+    await this.createComposerComponent(composer);
```

Comment:
Not your fault, don't change this in your PR, just something I wanted to point out.

`mail.composer` is an artificial split of concern of `mail.thread` at model level. Ideally it shouldn't have leaked in component... That was a mistake to have component that rely on `mail.composer`... Well, I think we wanted `mail.composer` to be the state of component in models, so something alike to `mail.thread_view`, but we couldn't made it at the time so we simplified with shared state among all threads.

Ok to let that for now, I'm expecting a cleaning of thread composer in the future.